### ISSUE TITLE
$extends doesn't work well in an array.

### ISSUE THREAD

**Issue by mikami-github** [dakusui/jq-front#60]
jq-front version: The latest version as of Nov 5, 2019

# Reproduce procedure
1. Prepare a jq-front file like follows.
test.json:
```
{
    "$local": {
        "nodeA": {
            "aa": "aa"
        },
        "nodeB": {
        }
    },
    "a": [
        {
            "$extends": ["nodeA"],
            "a": "a"
        }
    ]
}
```
2. Run jq-front with the above file.
```
./jq-front test.json
```

# Expected
```
{
    "a": [
        {
            "aa": "aa",
            "a": "a"
        }
    ]
}
```
# Actual
```
{
    "a": [
        {
            "a": "a"
        }
    ]
}
```

**Comment by dakusui** [comment]
This made me think quite a while, but eventually I believe I have to say that it's a designed behavior.
Because jq-front's inheritance is achieved by ```*``` operator of ```jq```, which causes exactly this.

If you apply ```*``` between two object nodes, its behavior is intuitive.

```
$ jq -n '{"a":"A"} * {"a":"a"}'
{
  "a": "a"
}
```

However it involves an array, it's a different story.

```
$ jq -n '{"a":[{"aa":"A"}]} * {"a":[{"xx":"HELLO!"}]}'
{
  "a": [
    {
      "xx": "HELLO!"
    }
  ]
}
```

Inside an array, ```jq``` always simply prefers an element which came from the latter not overlay them.
I suppose this behavior is inevitable because in case those two elements are of different types, you need to take the latter anyway.

**Comment by dakusui** [comment]
A possible workaround for this is to have an intermediate node. (.work.nodeA)
```
"$local": {
  "nodeX": {
    "x":"X"
  }
}
"work": {
  "nodeA": {
     "$extends": ["nodeX"]
  }
}
"P":[
   "eval:object:$(ref .work.nodeA)",
   "",
   ""
]
}
```
This behaviour is worth mentioning in the documentation (probably in FAQ?).
Thank you very much for pointing out.

**Comment by dakusui** [comment]
This behaviour was improved by the fix for #69
